File: src/B_1267/B_1267.java
package B_1267;

import java.io.*;
import java.util.StringTokenizer;

public class B_1267 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; ++i) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        int y = Y(arr);
        int m = M(arr);
        if(y == m){
            System.out.println("Y M "+ y);
        }else if (y<m){
            System.out.println("Y "+y);
        }else {
            System.out.println("M "+m);
        }

    }
    static int Y(int[] s){
        int res = 0;
        for (int v:
             s) {
            res += Math.floor(v/30.0)+1;
        }
        return res*10;
    }
    static int M(int[] s){
        int res = 0;
        for (int v:
                s) {
            res += Math.floor(v/60.0)+1;
        }
        return res*15;
    }

}

File: src/B_1267/B_1267.kt
package B_1267

import java.io.*
import java.util.*
import kotlin.math.floor

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n: Int = readLine().toInt()
    val arr = IntArray(n)
    val st = StringTokenizer(readLine())
    for (i in 0 until n) {
        arr[i] = st.nextToken().toInt()
    }
    val y = Y(arr)
    val m = M(arr)
    if (y == m) {
        println("Y M $y")
    } else if (y < m) {
        println("Y $y")
    } else {
        println("M $m")
    }
}
fun Y(s: IntArray): Int {
    var res = 0
    for (v in s) {
        res += (floor(v / 30.0) + 1).toInt()
    }
    return res * 10
}

fun M(s: IntArray): Int {
    var res = 0
    for (v in s) {
        res += (floor(v / 60.0) + 1).toInt()
    }
    return res * 15
}